Spanish localization

rpalmaotero opened the pull request:
This PR implements the Spanish localization of ICUBAM.

olivierteboul commented on icubam/backoffice/templates/icu.html:
```diff
@@ -9,7 +9,7 @@
         <div class="row mb-2">
           <div class="col-sm-6">
-            <h2>{{_("Manage your")}} {{_("ICUs")}}</h2>
+            <h2>{{ _('Manage ICUs') }}</h2>
```
I think this pattern occurs in other file. Why not keeping it the way it was ?

rpalmaotero replied:
Hi Olivier thanks for your comments. While rebasing my feature branch to master this conflict happened, I guess decomposing the translations would lead to DRY-er localization files. I'll update the branch.

olivierteboul replied:
thank you !

olivierteboul commented on icubam/backoffice/templates/icu.html:
```diff
@@ -20,100 +20,99 @@ <h2>{{_("Manage your")}} {{_("ICUs")}}</h2>
-                <input type="checkbox" name="is_active" {% if icu.is_active %} checked{% end %} data-bootstrap-switch
-                  data-off-color="danger" data-on-color="success">
-              </div>
+            value="{% if icu.icu_id is not None%}{{icu.icu_id}}{% end %}" </input> <div class="card-body">
```
break line after </input> would increase readability

olivierteboul commented on icubam/backoffice/templates/icu.html:
```diff
@@ -20,100 +20,99 @@ <h2>{{_("Manage your")}} {{_("ICUs")}}</h2>
 
+{% if error %}
+<div class="alert alert-danger" role="alert">
+  {{_('Failed to save ICU')}}
```
since you are on this. maybe we could save us some translations by having something like
`{{_('Error while saving ')}} {{_('ICU')}}`
It might not lead to perfect spanish or french, but it s only an error message, which appears elsewhere

rpalmaotero replied:
Sure, it goes in the same lines as the first comment. Will be updated.

olivierteboul commented on icubam/backoffice/templates/list.html:
```diff
@@ -100,6 +100,6 @@ <h3 class="card-title">{{_("Add")}}</h3>
 <script>
   let columns = {% raw columns %}
-  $(x => setDatatable("#my_data", columns))
+  $(x => setDatatable("#my_data", columns, "{{ _('dataTables.english.lang') }}"))
```
cool !

olivierteboul commented on icubam/backoffice/templates/login.html:
```diff
@@ -57,7 +57,7 @@
           </div>
           <div class="col-4">
-            <button type="submit" class="btn btn-primary btn-block">Sign In</button>
+            <button type="submit" class="btn btn-primary btn-block">{{ _('Sign In') }}</button>
```
good catch!

olivierteboul commented on icubam/backoffice/templates/map.html:
```diff
@@ -22,7 +22,7 @@
   let all_showed = false
   function initMap () {
-    plotMap(data, center)
+    plotMap(data, center, "{{ _('Tout montrer') }}", "{{ _('Clicker pour afficher') }}")
```
nice, but i would put those in English here and add two entries in fr.csv. This would be more consistent IMO

olivierteboul commented on icubam/backoffice/templates/operational-dashboard.html:
```diff
@@ -18,7 +18,7 @@ <h1 class='text-center'>{{ _("Operational dashboard") }}</h1>
 	     {{ _("Region") }}:
            {% end %}
-	   {{ current_region_name }}
+	   {{ _(current_region_name) }}
```
This is more debatable. I believe the region name does not need to be translated, does it?

rpalmaotero replied:
I agree. The reason of this l18n was to be able to localize "Toutes les régions" but giving it a second thought that could be made in the server handler. Will be updated.

olivierteboul commented on icubam/backoffice/templates/region.html:
```diff
@@ -9,7 +9,7 @@
         <div class="row mb-2">
           <div class="col-sm-6">
-            <h2>{{_("Manage your")}} {{_("Regions")}}</h2>
+            <h2>{{ _('Manage Regions') }}</h2>
```
i would stick to previous patterns too

olivierteboul commented on icubam/backoffice/templates/user.html:
```diff
@@ -35,8 +35,8 @@ <h3 class="card-title">{% if user.user_id %}{{_("Edit user")}}{% else %}{{_("Cre
             <input type="text" id="user_id" name="user_id" hidden
-              value="{% if user.user_id is not None%}{{user.user_id}}{% end %}" </input>
-            <!-- Required and does not accept names that contain only spaces. -->
+              value="{% if user.user_id is not None%}{{user.user_id}}{% end %}" </input> <!-- Required and does not
```
line break?

olivierteboul commented on icubam/backoffice/translations/es.csv:
```diff
@@ -0,0 +1,101 @@
+Taux de refus (covid) / capacité totale,Tasa de rechazo (COVID) / capacidad total
+Taux de remplissage lits (total),Tasa de ocupación de camas (total)
+Taux de remplissage lits (covid),Tasa de ocupación de camas (COVID)
+Fraction (%),Fracción (%)
```
awesome !

dulacarnold replied:
If there are french strings in the code, it would be better to translate them to english and have the translation file be en->es, and we will update the en->fr file, no?

rpalmaotero replied:
Yeah, for the sake of consistency I agree. I'll change that.

rpalmaotero commented:
@olivierteboul @dulacarnold hi guys, just updated the branch after your comments. I also modified the `BaseHandler.get_user_locale` method in order to consider the HTTP request `Accept-Language` header.

olivierteboul commented on icubam/backoffice/handlers/base.py:
```diff
@@ -52,9 +52,12 @@ def get_current_user(self):
-    return tornado.locale.get(locale) if locale else None
+    locale_code = self.get_query_argument('hl', default=None)
+    if locale_code is None:
+      return self.get_browser_locale()
```
cool

rpalmaotero commented:
@olivierteboul ok great, thanks. I've resolved the conflict and merge now.
